tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/jdt/ui/wizards/NewInterfaceCreationWizardTest.java

```diff
@@ -0,0 +1,277 @@
+		extendedIterfaceList.add("javax.accessibility.AccessibleStreamable");
+		extendedIterfaceList.add("javax.accessibility.AccessibleText");
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.addExtendedInterface(extendedInterfaceName);
+		}
+
+		assertTrue(
+				"List of interfaces obtained using function getExtendedInterfaces() has to be the same as list interfaces, that has been set.",
+				interfacePage.getExtendedInterfaces().equals(extendedIterfaceList));
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.removeExtendedInterface(extendedInterfaceName);
+		}
+
+		// cancel the wizard
+		interfaceDialog.cancel();
+	}
+
+	@Test
+	public void testAddingAndRemovingNonexistingAndNotFullySpecifiedInterfaces() {
+		// open the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceCreationWizardPage interfacePage = new NewInterfaceCreationWizardPage(interfaceDialog);
+
+		// try to add a non-existing extended interface
+		try {
+			interfacePage.addExtendedInterface("nonexisting.interface.name");
+			fail("RedDeer exception was not thrown when trying to add a non-existing extended interface.");
+		} catch (RedDeerException e) {
+			new PushButton("Cancel").click();
+		}
+
+		// try to add an extended interface, that is not fully specified by its name
+		try {
+			interfacePage.addExtendedInterface("Acl");
+			fail("RedDeer exception was not thrown when trying to add an extended interface, that is not fully specified by its name.");
+		} catch (RedDeerException e) {
+			new PushButton("Cancel").click();
+		}
+
+		// try to remove an extended interface, that is not on the list
+		try {
+			interfacePage.removeExtendedInterface("javax.accessibility.AccessibleAction");
+			fail("RedDeer exception was not thrown when trying to remove an extended interface, that is not on the list.");
+		} catch (RedDeerException e) {
+
+		}
+
+		// try to remove an added extended interface, that was not fully specified
+		interfacePage.addExtendedInterface("org.w3c.dom.views.AbstractView");
+		try {
+			interfacePage.removeExtendedInterface("AbstractView");
+			fail("RedDeer exception was not thrown when trying to remove an extended interface, that was added, but was not fully specified.");
+		} catch (RedDeerException e) {
+			interfacePage.removeExtendedInterface("org.w3c.dom.views.AbstractView");
+		}
+
+		// cancel the wizard
+		interfaceDialog.cancel();
+	}
+
+	private static void createTestProject() {
+		JavaProjectWizard dialog = new JavaProjectWizard();
+		dialog.open();
+		new NewJavaProjectWizardPageOne(dialog).setProjectName(PROJECT_NAME);
+		dialog.finish();
+	}
+
+	private static void deleteAllProjects() {
+		EditorHandler.getInstance().closeAll(false);
+		ProjectExplorer pe = new ProjectExplorer();
+		pe.open();
+		for (DefaultProject p : pe.getProjects()) {
+			try {
+				DeleteUtils.forceProjectDeletion(p, true);
```

Comment:
Is there a reason for using this function? You can use deleteAllProjects(boolean deleteFromFileSystem, TimePeriod timeout) https://github.com/eclipse/reddeer/blob/37f27affd709a9622d84e8d29d047dc78a0de1b4/plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/ui/navigator/resources/AbstractExplorer.java#L155

Reply:
No, there is no reason - I just did not realize there is more appropriate way to delete projects. Fixed.